### PR TITLE
Use Debugger.newSources events if emitted.

### PR DESCRIPTION
**Pull request by loganfsmyth**
We're not emitting these yet, but we're looking to switch over to them so this patch aims to support both.

Fixes https://linear.app/replay/issue/BAC-3460/use-debuggernewsources-in-devtools-codebase

### CODE REVIEW

File: packages/shared/utils/testing.ts [hunk 1]
```diff
@@ -126,6 +128,7 @@ export function createPointDescription({
 }
 
 export type SourceTuple = ["Debugger.newSource", newSource];
+export type SourcesTuple = ["Debugger.newSources", newSources];
 
 export function createSource({
```
Comment (loganfsmyth) [inline packages/shared/utils/testing.ts:133]: @bvaughn This testing stuff is what I wanted to ask you about. I'm not familiar with this code, so while I've done my best to add all this in parallel with what was already here, I'm not sure if there's any issues we should worry about here for the test fixture data, and how we manage that as we swap the the API over to the new event type.

File: packages/shared/client/ReplayClient.ts [hunk 2]
```diff
@@ -417,10 +418,17 @@ export class ReplayClient implements ReplayClientInterface {
...
       sources.push(source);
     };
+    const newSourcesListener = ({ sources }: newSources) => {
+      for (const source of sources) {
+        sources.push(source);
+      }
+    };
 
     client.Debugger.addNewSourceListener(newSourceListener);
+    client.Debugger.addNewSourcesListener(newSourcesListener);
```
Comment (bvaughn) [inline packages/shared/client/ReplayClient.ts:428]: What prevents us from adding every source twice? Is the expectation here that _only_ one of these two events will fire for any given session?

File: packages/shared/utils/testing.ts (hunk 1 above)
Comment (bvaughn) [reply packages/shared/utils/testing.ts:133]: As of #9107, there is no longer any fixture data being used by our screenshot tests.

File: packages/shared/client/ReplayClient.ts (hunk 2 above)
Comment (loganfsmyth) [reply packages/shared/client/ReplayClient.ts:428]: Yeah the API wouldn't emit both, just one or the other. I'm going to delete the other listener as soon as things is deployed, I just have both in this transition period.

File: packages/shared/utils/testing.ts (hunk 1 above)
Comment (loganfsmyth) [reply packages/shared/utils/testing.ts:133]: I don't think I have enough context to know how that relates to the question. Does that mean I'm fine to land this as-is?

File: packages/shared/client/ReplayClient.ts [hunk 3]
```diff
@@ -43,6 +43,7 @@ import {
   getTopFrameResult,
   keyboardEvents,
   navigationEvents,
+  newSources,
```
[the remark this answers is not in the document]
Comment (loganfsmyth) [reply packages/shared/client/ReplayClient.ts:46]: No argument there

File: packages/shared/client/ReplayClient.ts (hunk 2 above)
Comment (markerikson) [reply packages/shared/client/ReplayClient.ts:428]: @loganfsmyth note that the routines logic depends on fetching sources as well

Comment (loganfsmyth) [reply packages/shared/client/ReplayClient.ts:428]: Yup that should all be updated with the backend conversion https://github.com/replayio/backend/pull/7961

Comment (loganfsmyth) [reply packages/shared/client/ReplayClient.ts:428]: Woops wrong PR, this one: https://github.com/replayio/backend/pull/7958

**Review by bvaughn — APPROVED** [review]

**Pull request merged**